src/kernel/emulation/linux/misc/sysctl_machdep.c

```diff
@@ -51,3 +89,131 @@ sysctl_handler(handle_brand_string)
 	return 0;
 }
 
+sysctl_handler(handle_max_basic)
+{
+	setup(0);
+
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+
+
+
+	copyout_int(eax, (char*)old,oldlen);
+
+	return 0;
+}
+
+sysctl_handler(handle_family)
+{
+    setup(1);
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+    eax = eax >> 7;
+    eax &= 15;
+
+    copyout_string(old,"%d",eax);
+
+    return 0;
+}
+
+sysctl_handler(handle_model)
+{
+    setup(1);
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+    eax = eax >> 3;
+    eax &= 15;
+
+    copyout_int(eax,(char*)old, oldlen);
+
+    return 0;
+}
+
+sysctl_handler(handle_stepping)
+{
+    setup(1);
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+    eax &= 15;
+
+    copyout_int(eax,(char*)old, oldlen);
+
+    return 0;
+}
+
+sysctl_handler(handle_brand_string)
+{
+    setup(0x80000000);
+
+    __cpuid(level,eax,eabx, eacx, eadx);
+
+    if(eax < 0x80000004) // the information is not implemented
+        return 2;
+
+
+
+    union
+    {
+        unsigned int brand[12];
+        char name[49];
+    } v;
+
+		for (int i = 1; i < 4; i++)
+		{
+				eax = 0x80000000+i;
+				__cpuid(level, eax, ebx, ecx, edx);
+				v.brand[0x0+(i-1)*4] = eax;
+				v.brand[0x1+(i-1)*4] = ebx;
+				v.brand[0x2+(i-1)*4] = ecx;
+				v.brand[0x3+(i-1)*4] = edx;
+		}
+
+    v.name[48] = 0;
+    copyout_string(v.name, (char*) old, oldlen);
+
+    return 0;
+}
+sysctl_handler(handle_features)
+{
+
+    if(old != NULL)
+    {
+        setup(1);
+
+        char features[][] = {"FPU","VME", "DE", "PSE", "TSC", "MSR", "PAE", "MCE", "CX8", "APIC", "","SEP","MTRR","PGE",
+                            "MCA", "CMOV", "PAT", "PSE-36", "PSN", "CLFSH", "", "DS", "ACPI", "MMX", "FXSR", "SSE", "SSE2", "SS",
+                            "HTT", "TM","IA64","PBE"};
+
+        __cpuid(0,eax,ebx,ecx,edx);
+
+        int counter = 0;
+        int current_old_len = __simple_strlen(old);
+
+        for (int i = 0; i < 32; i++)
+        {
+
+            if(i == 10 || i == 20)
+                continue;
+
+            if(edx>>i&1 && current_old_len < oldlen)
+            {
+                int len = __simple_strlen(features[i]);
+                __simple_sprintf(old,"%s %s",old,features[i]);
```

Comment:
You're adding an extra space here when the string is still empty.
```suggestion
                if (current_old_len > 0)
                  __simple_sprintf(old,"%s %s",old,features[i]);
                else
                  __simple_sprintf(old,"%s",features[i]);
```